File: scripts/bundle.py
import uuid
import json


class Bundle:
    id = ""
    idPsp = ""
    name = ""
    description = ""
    paymentAmount = 0
    minPaymentAmount = 0
    maxPaymentAmount = 0
    paymentMethod = None
    touchpoint = None
    type = "GLOBAL"
    transferCategoryList = list()
    validityDateFrom = None
    validityDateTo = None

    counter = 0

    def __init__(self, counter=1, id_psp="12345678901", payment_amount_range=(0, 10000), type="GLOBAL", transferCategoryList=list(), payment_method='ANY', touchpoint='ANY'):
        self.id = str(uuid.uuid1())
        self.idPsp = id_psp
        self.counter = counter

        self.type = type

        self.name = "%s-bundle-%s" % (self.type.lower(), self.counter)
        self.description = "test description %s" % self.counter

        self.paymentAmount = self.counter + 100
        self.minPaymentAmount = payment_amount_range[0]
        self.maxPaymentAmount = payment_amount_range[1]

        self.paymentMethod = payment_method
        self.touchpoint = touchpoint

        self.transferCategoryList = transferCategoryList

    def __repr__(self):
        return json.dumps(self.__dict__)

File: src/main/java/it/gov/pagopa/afm/utils/entity/Bundle.java
package it.gov.pagopa.afm.utils.entity;

import com.azure.spring.data.cosmos.core.mapping.Container;
import com.azure.spring.data.cosmos.core.mapping.GeneratedValue;
import com.azure.spring.data.cosmos.core.mapping.PartitionKey;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.List;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Size;
import lombok.*;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.LastModifiedDate;

@Container(containerName = "bundles")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder(toBuilder = true)
public class Bundle {

  @Id @GeneratedValue private String id;

  @PartitionKey
  @NotNull
  @Size(max = 35)
  private String idPsp;

  @NotNull private String idChannel;

  @NotNull private String idBrokerPsp;

  private String idCdi;

  @NotNull private String abi;

  private Boolean digitalStamp;

  // true if bundle must be used only for digital stamp
  private Boolean digitalStampRestriction;

  private String name;
  private String description;

  private Long paymentAmount;
  private Long minPaymentAmount;
  private Long maxPaymentAmount;

  private String paymentType;

  private String touchpoint;

  private String type;

  private List<String> transferCategoryList;

  private LocalDate validityDateFrom;

  private LocalDate validityDateTo;

  @CreatedDate private LocalDateTime insertedDate;

  @LastModifiedDate private LocalDateTime lastUpdatedDate;
}
